File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    my_stack = [] 
    result = 0
    
    for i in ingredient:
        my_stack.append(i)     
        if len(my_stack) < 4: continue
        if i == 1 and my_stack[-4:] == [1, 2, 3, 1]:
            for j in range(4):
                my_stack.pop()
            result += 1
    return result
    
            

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
const solution = (ingredient) => {
    const ingredient_stack = [];
    let result = 0;
    ingredient.forEach((i) => {
        ingredient_stack.push(i);
        if (ingredient_stack.length >= 4) {
            if (ingredient_stack.slice(-4).join('') === '1231') {
                for (let i=0; i<4; i++) {
                    ingredient_stack.pop();
                }
                result += 1;
            }
        }
    });
    return result;
}